diff --git b/policer_main.c a/policer_main.c
index 549aea75..768c4f32 100644
--- b/policer_main.c
+++ a/policer_main.c
@@ -28,10 +28,9 @@ struct nf_config config;
 struct State *dynamic_ft;
 
 int policer_expire_entries(vigor_time_t time) {
+  assert(time >= 0); // we don't support the past
   vigor_time_t exp_time =
       VIGOR_TIME_SECONDS_MULTIPLIER * config.burst / config.rate;
-  if (time < exp_time) // we don't support the past
-    return 0;
   uint64_t time_u = (uint64_t)time;
   // OK because time >= config.burst / config.rate >= 0
   vigor_time_t min_time = time_u - exp_time;
