--- a/drivers/net/ixgbe/ixgbe_ethdev.c	
+++ b/drivers/net/ixgbe/ixgbe_ethdev.c
@@ -1318,10 +1318,11 @@
 	TAILQ_INIT(&filter_info->fivetuple_list);
 
 	/* initialize flow director filter list & hash */
-	ixgbe_fdir_filter_init(eth_dev);
+	// filter_init functions are not necessary and take a lot of time to verify.
+	//ixgbe_fdir_filter_init(eth_dev);
 
 	/* initialize l2 tunnel filter list & hash */
-	ixgbe_l2_tn_filter_init(eth_dev);
+	//ixgbe_l2_tn_filter_init(eth_dev);
 
 	/* initialize flow filter lists */
 	ixgbe_filterlist_init();
