--- dpdk-20.05-ref/lib/librte_eal/common/eal_common_memory.c	2020-05-26 15:41:39.000000000 +0000
+++ dpdk/lib/librte_eal/common/eal_common_memory.c	2020-07-23 09:01:46.058019056 +0000
@@ -48,7 +48,8 @@
 #error "madvise doesn't support this OS"
 #endif
 
-#define MAX_MMAP_WITH_DEFINED_ADDR_TRIES 5
+// 1 try is enough and it makes validation faster.
+#define MAX_MMAP_WITH_DEFINED_ADDR_TRIES 1
 void *
 eal_get_virtual_area(void *requested_addr, size_t *size,
 		size_t page_sz, int flags, int mmap_flags)
