diff --git a/config/common_base b/config/common_base
index 7830535..f908cbb 100644
--- a/config/common_base
+++ b/config/common_base
@@ -42,7 +42,8 @@ CONFIG_RTE_TOOLCHAIN=
 #
 # Use intrinsics or assembly code for key routines
 #
-CONFIG_RTE_FORCE_INTRINSICS=n
+# Enable atomic intrinsics, otherwise it uses inline ASM which KLEE doesn't support
+CONFIG_RTE_FORCE_INTRINSICS=y
 
 #
 # Machine forces strict alignment constraints.

@@ -83,7 +83,8 @@ CONFIG_RTE_MAX_LCORE=128
CONFIG_RTE_MAX_NUMA_NODES=8
CONFIG_RTE_MAX_HEAPS=32
-CONFIG_RTE_MAX_MEMSEG_LISTS=64
+# We do not need as much memory, reducing it makes validation faster
+CONFIG_RTE_MAX_MEMSEG_LISTS=1
# each memseg list will be limited to either RTE_MAX_MEMSEG_PER_LIST pages
# or RTE_MAX_MEM_MB_PER_LIST megabytes worth of memory, whichever is smaller
CONFIG_RTE_MAX_MEMSEG_PER_LIST=8192
CONFIG_RTE_MAX_MEM_MB_PER_LIST=32768
